### PR TITLE
Node.d.ts: Update definitions for module "tls"

### PR DESCRIPTION
**Pull request by hriss95**
case 2. Improvement to existing type definition.

This pull request updates the two interfaces `TlsOptions` and `ConnectionOptions` by introducing new properties and updating the types of already existing ones. Also changed the type of one of the parameters of the function `connect(options[, secureConnectionListener])` as it was using TlsOptions as a type and should use ConnectionOptions. All the changes are made according to the Node's [documentation](https://nodejs.org/dist/latest-v6.x/docs/api/tls.html).

The description of TlsOptions can be found in the documentation for function [createServer](https://nodejs.org/dist/latest-v6.x/docs/api/tls.html#tls_tls_createserver_options_secureconnectionlistener).

The description of ConnectionOptions can be found in the documentation for function [connect](https://nodejs.org/dist/latest-v6.x/docs/api/tls.html#tls_tls_connect_options_callback).

### CODE REVIEW

**Comment by hriss95** [comment]
Update: Reverted the changes on function `connect` as there was a single test failing. However, I think the test should be reviewed as it passes an object of type tlsOptions as an argument and I suspect the test predates the introduction of interface ConnectionOption. Although `ConnectionOptions` and `tlsOptions` have a lot of common properties they still are different types and the correct type to be passed to the connect function is `ConnectionOptions`. Let me know if you agree with that and I will change the test in question.

File: node/node.d.ts [hunk 1]
```diff
@@ -1993,32 +1993,47 @@ declare module "tls" {
     export interface TlsOptions {
         host?: string;
         port?: number;
-        pfx?: any;   //string or buffer
-        key?: any;   //string or buffer
+        pfx?: string | Buffer[];
```
Comment (vvakame) [inline node/node.d.ts]: > pfx <string> | <Buffer>

`pfx?: string | Buffer;` ?

File: node/node.d.ts [hunk 2]
```diff
@@ -1993,32 +1993,47 @@ declare module "tls" {
...
+        key?: string | string[] | Buffer | Object[];
```
Comment (vvakame) [inline node/node.d.ts]: > key &lt;string> | &lt;string[]> | &lt;Buffer> | &lt;Buffer[]> 

`key?: string | string[] | Buffer | Buffer[];` ?

**Comment by vvakame** [comment]
ah... I understand.
We should split interface depending on usecase.

**Comment by hriss95** [comment]
I am not sure I understand. Could you comment more on what you mean by splitting the interface depending on usecase? I am not really familiar with the usage of the module and just copied the types from the documentation as they are stated.

File: node/node.d.ts (hunk 2 above)
Comment (vvakame) [inline node/node.d.ts]: >  an array of objects in the format {pem: key, passphrase: passphrase}

We should use `string | string[] | Buffer | any[];`

**Comment by vvakame** [comment]
oh... I'm so sorry. 
It looks like I had misunderstood.

> I am not really familiar with the usage of the module

me too. 😉 

`export function connect(options: TlsOptions, secureConnectionListener?: () => void): ClearTextStream;` should be `export function connect(options: ConnectionOptions, secureConnectionListener?: () => void): ClearTextStream;`?

**Comment by hriss95** [comment]
Oh, yes, it should! That is what I was referring to in the description of the pull request. One of the tests contains a call to connect and passes an object of type TlsOptions to it and I was not passing the  Travis CI test. I will do another commit which alters the test.

**Comment by hriss95** [comment]
It should be fine now.

**Comment by vvakame** [comment]
thank you so much!

**Pull request merged**